src/Simulator/IncrementalRedstoneSimulator.h

```diff
@@ -107,126 +98,171 @@ class cIncrementalRedstoneSimulator :
+
 	/** Handles tripwire hooks
 	Performs correct meta and power setting for self by going in the direction it faces and looking for a continous line of tripwire bounded by another oppositely facing hook
 	If this line is complete, it verifies that at least on wire reports an entity is on top (via its meta), and performs its task
 	*/
 	void HandleTripwireHook(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
+
 	/** Handles trapped chests */
 	void HandleTrappedChest(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
-	/* ==================== */
 
-	/* ====== CARRIERS ====== */
+	
 	/** Handles redstone wire */
 	void HandleRedstoneWire(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
+
 	/** Handles repeaters */
 	void HandleRedstoneRepeater(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ, BLOCKTYPE a_MyState);
-	/* ====================== */
 
-	/* ====== DEVICES ====== */
+	/** Handles comparators */
+	void HandleRedstoneComparator(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
+	
+	
 	/** Handles pistons */
 	void HandlePiston(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
+
 	/** Handles dispensers and droppers */
 	void HandleDropSpenser(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
+
 	/** Handles TNT (exploding) */
 	void HandleTNT(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
+
 	/** Handles redstone lamps */
 	void HandleRedstoneLamp(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ, BLOCKTYPE a_MyState);
+
 	/** Handles doords */
 	void HandleDoor(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
+
 	/** Handles command blocks */
 	void HandleCommandBlock(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
+
 	/** Handles activator, detector, and powered rails */
 	void HandleRail(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ, BLOCKTYPE a_MyType);
+
 	/** Handles trapdoors */
 	void HandleTrapdoor(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
+
 	/** Handles fence gates */
 	void HandleFenceGate(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
+
 	/** Handles noteblocks */
 	void HandleNoteBlock(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
+
 	/** Handles tripwires */
 	void HandleTripwire(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ);
-	/* ===================== */
 
-	/* ====== Helper functions ====== */
+
 	/** Marks a block as powered */
-	void SetBlockPowered(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ, int a_RelSourceX, int a_RelSourceY, int a_RelSourceZ, unsigned char a_PowerLevel = MAX_POWER_LEVEL);
+	void SetBlockPowered(Vector3i a_RelBlockPosition, Vector3i a_RelSourcePosition, unsigned char a_PowerLevel = MAX_POWER_LEVEL);
+	void SetBlockPowered(int a_RelBlockX, int a_RelBlockY, int a_RelBlockZ, int a_RelSourceX, int a_RelSourceY, int a_RelSourceZ, unsigned char a_PowerLevel = MAX_POWER_LEVEL) { SetBlockPowered(Vector3i(a_RelBlockX, a_RelBlockY, a_RelBlockZ), Vector3i(a_RelSourceX, a_RelSourceY, a_RelSourceZ), a_PowerLevel); }
```

Comment:
That wont happen. Seriously you need to finish any refactors in this PR, unless its a public method thats used in lots of places.